app/Models/Event.php

```
@@ -0,0 +1,26 @@
+{
+    /** @use HasFactory<\Database\Factories\EventFactory> */
+    use HasFactory, HasUlids;
+
+    use Helpers\WithSchedule;
+
+    /**
+     * @return BelongsToMany<Person, Event, Participation, 'participation'>
+     */
+    public function participants(): BelongsToMany
+    {
+        return $this->belongsToMany(Person::class, Participation::class, relatedPivotKey: 'participant_id')
```

PHPStan (2.1.15)

Method App\Models\Event::participants() should return Illuminate\Database\Eloquent\Relations\BelongsToMany<App\Models\Person, App\Models\Event, App\Models\Participation, 'participation'> but returns Illuminate\Database\Eloquent\Relations\BelongsToMany<App\Models\Person, $this(App\Models\Event), Illuminate\Database\Eloquent\Relations\Pivot, 'participation'>.
Template type TDeclaringModel on class Illuminate\Database\Eloquent\Relations\BelongsToMany is not covariant. Learn more: https://phpstan.org/blog/whats-up-with-template-covariant

(return.type)